### PR TITLE
extension: provide more informations in `extension.FunctionContext`

### LINKED ISSUE #39049: provide more informations in `extension.FunctionContext`

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #39049

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: extension/function_test.go [hunk 1]
```diff
@@ -142,7 +192,7 @@ func TestFuncDynamicArgLen(t *testing.T) {
 	require.EqualError(t, tk.ExecToErr("select dynamic_arg_func(1, 2)"), expectedErrMsg)
 }
 
-func TestRegisterFunc(t *testing.T) {
+func TestRegisterExtensionFunc(t *testing.T) {
 	defer func() {
 		extension.Reset()
 	}()
```
Comment (xhebox) [inline extension/function_test.go:198]: Guess you can modify this defer, too.

**Review by xhebox — APPROVED** [review]

File: extension/function_test.go (hunk 1 above)
Comment (lcwangchao) [reply extension/function_test.go]: done

**Pull request merged**